src/Build/BuildCheck/API/BuildAnalyzerRule.cs

```diff
@@ -45,7 +49,7 @@ public class BuildAnalyzerRule
     /// <summary>
     /// Message format that will be used by the actual reports (<see cref="BuildCheckResult"/>) - those will just supply the actual arguments.
     /// </summary>
-    public string MessageFormat { get; }
+    public string MessageFormat { get; set; }
```

Comment:
@JanKrivanek , heads up on the change

Reply:
I like the nicely readable and organized formats per both verbosity options. On the other hand adding a mutability to this class feels to high tradeoff.
Let's keep this immutable at the cost of less nice format (can be '{0}'). We can figure out better approach later onif needed often.